MCR 3193 fix pmd issues in mycore mets

Antoniafriedrich opened the pull request:
[Link to jira](https://mycore.atlassian.net/browse/MCR-3193).

Commit:
MCR-3193 UnnecessaryFullyQualifiedName

Commit:
MCR-3193 UseEqualsToCompareStrings

Commit:
MCR-3193 SwitchStmtsShouldHaveDefault

Commit:
MCR-3193 SimplifyBooleanReturns

Commit:
MCR-3193 CompareObjectsWithEquals

Commit:
MCR-3193 AvoidReassigningParameters

Commit:
MCR-3193 AvoidThrowingRawExceptionTypes

Commit:
MCR-3193 NPathComplexity

sebhofmann commented on mycore-mets/src/main/java/org/mycore/mets/model/converter/MCRAltoLinkTypeAdapter.java:
```diff
@@ -52,10 +53,12 @@ public MCRMetsAltoLink read(JsonReader jsonReader) throws IOException {
                 case "begin" -> begin = jsonReader.nextString();
-                case "end" -> end = jsonReader.nextString();
+                case "end"-> end = jsonReader.nextString();
+                default -> throw new JsonParseException(name + " not recognized");
```
previously unknown names were just ignored.

sebhofmann commented on mycore-mets/src/main/java/org/mycore/mets/model/converter/MCRMetsLinkTypeAdapter.java:
```diff
@@ -48,16 +49,19 @@ public MCRMetsLink read(JsonReader jsonReader) throws IOException {
+            switch (name) {
                 case "from" -> ml.setFromString(jsonReader.nextString());
                 case "to" -> ml.setToString(jsonReader.nextString());
+                default -> throw new JsonParseException(name + " is not a valid link type");
```
previously unknown names were just ignored.

sebhofmann reviewed (changes requested):
Some minor issues. Rest looks good.

Commit:
Change Switch cases to If Statements

Since the switch had no default and no default was wanted it was replaced with if else.

Antoniafriedrich committed:
Merge branch 'main' into issues/MCR-3193-fix-pmd-issues-in-mycore-mets

Antoniafriedrich committed:
Merge branch 'main' into issues/MCR-3193-fix-pmd-issues-in-mycore-mets

rsteph-de commented on mycore-mets/src/main/java/org/mycore/mets/model/converter/MCRAltoLinkTypeAdapter.java:
```diff
@@ -52,10 +52,13 @@ public MCRMetsAltoLink read(JsonReader jsonReader) throws IOException {
-            switch (jsonReader.nextName()) {
-                case "altoFile" -> fileID = jsonReader.nextString();
-                case "begin" -> begin = jsonReader.nextString();
-                case "end" -> end = jsonReader.nextString();
```
We want to keep the switch and add a default branch

rsteph-de commented on mycore-mets/src/main/java/org/mycore/mets/model/converter/MCRMetsLinkTypeAdapter.java:
```diff
@@ -48,16 +48,19 @@ public MCRMetsLink read(JsonReader jsonReader) throws IOException {
         while (jsonReader.hasNext()) {
-            switch (jsonReader.nextName()) {
-                case "from" -> ml.setFromString(jsonReader.nextString());
-                case "to" -> ml.setToString(jsonReader.nextString());
```
We want to keep the switch and add a default branch

Commit:
revert if to switches and add default

Antoniafriedrich committed:
Merge branch 'main' into issues/MCR-3193-fix-pmd-issues-in-mycore-mets

rsteph-de commented:
2024-10-15T07:13:53.4502745Z 07:13:53,449 [INFO] There are 3 errors reported by Checkstyle 10.12.7 with /home/runner/work/mycore/mycore/checkstyle.xml ruleset.
2024-10-15T07:13:53.4527722Z 07:13:53,451 [ERROR] src/main/java/org/mycore/mets/model/converter/MCRAltoLinkTypeAdapter.java:[55,43] (coding) InnerAssignment: Inner assignments should be avoided.
2024-10-15T07:13:53.4531927Z 07:13:53,451 [ERROR] src/main/java/org/mycore/mets/model/converter/MCRAltoLinkTypeAdapter.java:[56,39] (coding) InnerAssignment: Inner assignments should be avoided.
2024-10-15T07:13:53.4536340Z 07:13:53,451 [ERROR] src/main/java/org/mycore/mets/model/converter/MCRAltoLinkTypeAdapter.java:[57,35] (coding) InnerAssignment: Inner assignments should be avoided.

rsteph-de commented on mycore-mets/src/main/java/org/mycore/mets/model/converter/MCRAltoLinkTypeAdapter.java:
```diff
@@ -51,13 +51,14 @@ public MCRMetsAltoLink read(JsonReader jsonReader) throws IOException {
 
         jsonReader.beginObject();
         while (jsonReader.hasNext()) {
-            //CSOFF: InnerAssignment
```
the 2 comments need to be added to disable checkstyle here

Commit:
adds back checkstyle comments

rsteph-de reviewed (approved):
Looks good for me now

rsteph-de committed:
Merge branch 'main' into issues/MCR-3193-fix-pmd-issues-in-mycore-mets

rsteph-de committed:
Merge branch 'main' into issues/MCR-3193-fix-pmd-issues-in-mycore-mets
